### DISCUSSION TITLE
Convenient way to initialize a state with data

### DISCUSSION THREAD

**Discussion by johnrudolph** [hirethunk/verbs#38] [Ideas]
I would love a way to initialize a state, pass in whatever data I want into its variables, and then snapshot it so that I can have whatever starting point I want for a given state. 

The main place this would be helpful for me would be in tests. For example, my dream situation would be if I could do something like this: 

![Screenshot 2023-12-10 at 10 24 10 PM](https://github.com/hirethunk/verbs/assets/18694510/74260855-3457-4913-aba4-babd6e5eb2c2)

I also think this could be useful in other contexts like migrating a legacy system to event sourcing, where you need to create states for things that already exist and aren't starting fresh. You could essentially seed data into these states, snapshot them, and then go from there.

**Comment by inxilpro** [comment]
I think maybe the move would be two separate approaches for tests and production code. In tests, I think we should have the concept of state factories, so that you can get your state set up however you need for the tests without having to fire a bunch of events. For setting up state in production, though, I think it's really important that an event is fired—snapshots shouldn't be relied upon. So what if we had an internal `StateInitialized` event that you could fire?
